services/management/backup/artifacts_service.go

```diff
@@ -96,6 +100,104 @@ func (s *ArtifactsService) ListArtifacts(context.Context, *backupv1beta1.ListArt
 	}, nil
 }
 
+func (s *ArtifactsService) deleteArtifactRequirements(
+	artifactID string,
+) (string, *models.S3LocationConfig, error) {
+	var s3Config *models.S3LocationConfig
+	var artifactName string
+	if err := s.db.InTransaction(func(tx *reform.TX) error {
+		artifact, err := models.FindArtifactByID(tx.Querier, artifactID)
+		switch {
+		case err == nil:
+		case errors.Is(err, models.ErrNotFound):
+			return status.Errorf(codes.NotFound, "Artifact with ID %q not found.", artifactID)
+		default:
+			return err
+		}
+		if artifact.Status != models.SuccessBackupStatus && artifact.Status != models.ErrorBackupStatus {
+			return status.Errorf(codes.FailedPrecondition, "Artifact with ID %q isn't in the final state.", artifactID)
+		}
+
+		artifactName = artifact.Name
+
+		restoreItems, err := models.FindRestoreHistoryItems(tx.Querier, &models.RestoreHistoryItemFilters{
```

Comment:
can we just query restore items with status == in progress

Reply:
yes, we can, I will do

Reply:
status filter added